pkg/landscaper/installations/executions/template/gotemplate.go

```diff
@@ -480,3 +483,68 @@ func resolveComponents(defaultCD *cdv2.ComponentDescriptor, list *cdv2.Component
 
 	return components, nil
 }
+
+func getImageVectorGoFunc(cd *cdv2.ComponentDescriptor, list *cdv2.ComponentDescriptorList) func(args ...interface{}) map[string]interface{} {
+	return func(args ...interface{}) map[string]interface{} {
+		internalCd := cd
+		internalComponents := list
+
+		if len(args) > 2 {
+			panic("Too many arguments for getImageVectorGoFunc")
+		} else if len(args) == 1 {
+			data, err := json.Marshal(args[0])
+			if err != nil {
+				panic("Unable to marshall first argument to getImageVectorGoFunc")
+			}
+
+			internalCd = &cdv2.ComponentDescriptor{}
+			if err = codec.Decode(data, internalCd); err != nil {
+				panic("Unable to marshall first argument to component descriptor.")
+			}
+		} else if len(args) == 2 {
+			data, err := json.Marshal(args[0])
+			if err != nil {
+				panic("Unable to marshall first argument to component descriptor.")
+			}
+
+			internalCd = &cdv2.ComponentDescriptor{}
+			if err := codec.Decode(data, internalCd); err != nil {
+				panic("Unable to marshall first argument to component descriptor.")
+			}
+
+			componentsData, err := json.Marshal(args[1])
+			if err != nil {
+				panic("Unable to marshall second argument to component descriptor list.")
+			}
+
+			internalComponents = &cdv2.ComponentDescriptorList{}
+			if err := codec.Decode(componentsData, internalComponents); err != nil {
+				panic("Unable to marshall second argument to component descriptor list.")
+			}
```

Comment:
could be simplified
```suggestion
		} else  {
			data, err := json.Marshal(args[0])
			if err != nil {
				panic("Unable to marshal first argument to getImageVectorGoFunc")
			}

			internalCd = &cdv2.ComponentDescriptor{}
			if err = codec.Decode(data, internalCd); err != nil {
				panic("Unable to marshal first argument to component descriptor.")
			}
		        if len(args) == 2 {
			  componentsData, err := json.Marshal(args[1])
			  if err != nil {
				  panic("Unable to marshal second argument to component descriptor list.")
			  }

			  internalComponents = &cdv2.ComponentDescriptorList{}
			  if err := codec.Decode(componentsData, internalComponents); err != nil {
				panic("Unable to marshal second argument to component descriptor list.")
			  }
			}
```

Reply:
This approach is more simple but it does not allow to provide only a ComponentDescriptorList and no CD. If this is no valid use case we should take this alternative.

Reply:
We have simplified the code. The behaviour is unchanged: the template function can be used either without arguments or with one argument (the component descriptor), or with two arguments (component descriptor and components list).